Added documentation about the different control net file versions

jessemapel opened the pull request:
The documentation was added to ControlNetVersioner's class documentation. Relevant parts should also be copied to the documentation for the ControlPointV#### classes.

Also changed some formatting in the rest of ControlNetVersioner's class documentation.

jessemapel committed:
Added documentation about the different control net file version to ControlNetVersioner's documentation.

jessemapel commented:
![](https://github.trello.services/images/mini-trello-icon.png) [explicitly document the changes from v1 -> 2 -> 3 -> 4 ->5 somewhere](https://trello.com/c/4coBZIVd/127-explicitly-document-the-changes-from-v1-2-3-4-5-somewhere)

jessemapel commented:
If I get to the documentation for the ControlPointV#### classes, I will include that in this PR

jessemapel commented:
Any editing feedback is certainly welcome!

jessemapel committed:
improved ControlPointV0001 class documentation

jessemapel committed:
Improved ControlPointV0002 class documentation

jessemapel committed:
improved ControlPointV0003 class documentation

jessemapel commented:
Documentation for the contents of the versioned control point classes has been updated. This is all set to go now.

krlberry commented on isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.h:
```diff
@@ -85,56 +93,66 @@ namespace Isis {
+   *   </li>
+   *   <li>Write a Pvl header into the original blank header at the start of the
    *        file. This header contains a flag indicating the network is a
    *        protobuf formatted network, the version of the format, the byte
```
I found this wording slightly confusing. Maybe add a colon after `This header contains`? This would make it more clear that it needs to be read as "this header contains: blah, blah, blah" and not "this header contains a flag indicating: blah, blah blah."

krlberry commented on isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.h:
```diff
@@ -144,18 +162,199 @@ namespace Isis {
+   *
+   * <h3>Control Network File Format History</h3>
+   *
+   * Prior to the creation of this versioning class, which was release with
```
s/release/released?

krlberry commented on isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.h:
```diff
@@ -144,18 +162,199 @@ namespace Isis {
+   *     that contains offsets to the binary components of the file. This
+   *     header may also contain general information about the control network.
+   *     This information is only for user reference is not used when reading
+   *     the file.
```
Don't we use the offsets to the binary components (from the Pvl header) when reading the file?

krlberry replied:
And I think we also use the version number in the header to determine which readProtobufV#### to use.

krlberry commented on isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.h:
```diff
@@ -144,18 +162,199 @@ namespace Isis {
+   *     contains the majority of the network data. This is a hierarchical
+   *     structure with general network information such as the network
+   *     description at the top level. Below that is the control point
+   *     information. Lastly, is the control measure information. This
```
Suggest changing "Lastly, is the control measure information" to something like "Beneath that is the control measure information." Or something else. That one sentence doesn't sound completely right to me for some reason.

krlberry commented on isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.h:
```diff
@@ -144,18 +162,199 @@ namespace Isis {
+   *     the Protobuf Core which contains all of the control point and control
+   *     measure information. This is structured as consecutive protobuf
+   *     messages where each message contains all of the information for a
+   *     control point and its control measures. Because, protobuf messages
```
Remove the comma after `Because,`

krlberry commented on isis/src/control/objs/ControlNetVersioner/ControlPointV0002.h:
```diff
@@ -37,17 +37,122 @@ namespace Isis {
+   *       ground point in meters</li>
+   *   <li><em>AdjustedZ:</em> The body fixed Z coordinate of the adjusted
+   *       ground point in meters</li>
+   *   <li><em>LatitudeConstrained:</em> If the latitude of the grount point
```
s/grount/ground

krlberry commented on isis/src/control/objs/ControlNetVersioner/ControlPointV0002.h:
```diff
@@ -37,17 +37,122 @@ namespace Isis {
+   *       ground point in meters</li>
+   *   <li><em>LatitudeConstrained:</em> If the latitude of the grount point
+   *       is constrained</li>
+   *   <li><em>LongitudeConstrained:</em> If the longitude of the grount point
```
s/grount/ground

krlberry commented on isis/src/control/objs/ControlNetVersioner/ControlPointV0002.h:
```diff
@@ -37,17 +37,122 @@ namespace Isis {
+   *       is constrained</li>
+   *   <li><em>LongitudeConstrained:</em> If the longitude of the grount point
+   *       is constrained</li>
+   *   <li><em>RadiusConstrained:</em> If the radius of the grount point
```
s/grount/ground

krlberry commented on isis/src/control/objs/ControlNetVersioner/ControlPointV0003.h:
```diff
@@ -40,8 +40,122 @@ namespace Isis {
-   * this class works with both versions. The differences between the version 3 and 4 Pvl
-   * serialization is small enough that it is handled in the PvlObject constructor.
+   * this class works with both versions. The difference between the version 3 and 4 Pvl
+   * serializations is changing Ground and Tie points to Fixed and Free respectively. This i
```
s/i/is

krlberry commented on isis/src/control/objs/ControlNetVersioner/ControlPointV0003.h:
```diff
@@ -40,8 +40,122 @@ namespace Isis {
+   *       ground point in meters</li>
+   *   <li><em>AdjustedZ:</em> The body fixed Z coordinate of the adjusted
+   *       ground point in meters</li>
+   *   <li><em>LatitudeConstrained:</em> If the latitude of the grount point
```
s/grount/ground

krlberry commented on isis/src/control/objs/ControlNetVersioner/ControlPointV0003.h:
```diff
@@ -40,8 +40,122 @@ namespace Isis {
+   *       ground point in meters</li>
+   *   <li><em>LatitudeConstrained:</em> If the latitude of the grount point
+   *       is constrained</li>
+   *   <li><em>LongitudeConstrained:</em> If the longitude of the grount point
```
s/grount/ground

krlberry commented on isis/src/control/objs/ControlNetVersioner/ControlPointV0003.h:
```diff
@@ -40,8 +40,122 @@ namespace Isis {
+   *       is constrained</li>
+   *   <li><em>LongitudeConstrained:</em> If the longitude of the grount point
+   *       is constrained</li>
+   *   <li><em>RadiusConstrained:</em> If the radius of the grount point
```
s/grount/ground

krlberry reviewed (commented):
This is great! So much good information in here. My comments are mostly about typos and also contain a couple of suggestions for rewording sentences. 

I used `s/original/replacement` for comments that were just replace `original` with `replacement` because it was easier than writing it out in english.

jessemapel committed:
Fixed typos in ControlNetVersioner documentation

jessemapel committed:
Fixed typos in ControlPointV0002 documentation

jessemapel committed:
Fixed typo in ControlPointV0003 docs

Also reworded a little bit to read better

jessemapel commented:
Review feedback has been incorporated:

- Reworded how the Pvl header is described for version 2 binary files.
- Changed the description of the version 1/2 Protobuf Core to be clearer
- Reworded the explanation of v3 -> v4 Pvl differences in ControlPointV0003
- Fixed typos

jessemapel commented:
Also, I didn't mention anywhere in the ControlNetVersioner documentation how it determines if a file is Pvl v Protobuf and which version it is. I just said that it figures those out somehow. Should it be included in the documentation and if so where?

krlberry reviewed (approved):
:+1: Looks good!

krlberry commented:
@jessemapel Hmm... that's a good question. I think it is okay as is. It's straightforward from looking at the code in `ControlNetVersioner::read` how the pvl vs protobuf determination is made, and the existing `<li>Determine if the network is stored in Pvl or protobuf format</li>` is probably enough detail for the documentation. If you wanted to add more information, somewhere near that line would probably make sense.
